### ISSUE TITLE
(3.3.1) FormBuilderDateTimePicker shows keyboard if editable: true

### ISSUE THREAD

**Issue by abhinavc** [danvick/flutter_form_builder#107]
Hi Danvick, 
I recently upgraded Flutter and packages in my project, which upgraded `flutter_form_builder` from version `3.0.1` to `3.3.1`. And now I see a weird issue with the date picker. 

I have the following in my code
```dart
FormBuilderDateTimePicker(
  .... 
  inputType: InputType.date,
  editable: false,
  initialValue: null, 
  initialDate: DateTime.parse("19800615"),
  resetIcon: Icons.undo,
  validators: [...],
);
```
In `3.0.1` setting `editable: false` would **not** disable the field but allowed edits only through the date picker - which was fine. 

But in `3.3.1`, `editable: false` disables the form field entirely - including the `resetIcon`
Setting `editable: true`, however, enables the field but shows

- the keyboard if some date set 
- date picker only if form field is empty (like after `resetIcon` is pressed)

Setting `enabled: true, editable: false` seems to have no effect (the field stays disabled) 

I tried downgrading to `3.0.1`. But the interplay with the other upgraded packages now creates an on-screen glitch that I didn't see before. 

I think only one of `editable` and `enabled` is required. If `enabled: true`, then edit should only be allowed through a date-picker. 

Not sure if [datetime_picker_formfield#38](https://github.com/jifalops/datetime_picker_formfield/issues/38) is relevant.

If this is a real/valid issue, then could you please provide a hot-fix release.

Thanks 
Abhinav

**Comment by abhinavc** [comment]
Ok, so it seems what is `editable` in `FormBuilderDateTimePicker` is `enabled` for `DateTimeField`.
`enabled` in `FormBuilderDateTimePicker` is redundant. 

So, if the aim is to allow edits only using date/time pickers (`onShowPicker`), then the issue is why date/time pickers don't show *when the form field get focus but is not empty*

It is possible to reset the field so that date/time pickers then show. But as a user, my expectation is 
to see the same input method whether the field is empty or not.

**Comment by danvick** [comment]
Hi @abhinavc,
Thanks for your contribution. It seems that in the later versions of `datetime_picker_formfield`, the `editable` field was removed. 
I'll be working to make sure that the input behaves as you've suggested.

**Comment by danvick** [comment]
Kindly try out the newly published v3.3.3

**Comment by abhinavc** [comment]
Hi Danvick,
Thanks for the fix. Works as expected in 3.3.4. 
Have another issue to report though :-)
